js/walutomat.js

```diff
@@ -0,0 +1,285 @@
+                    ],
+                },
+                'private': {
+                    'get': [
+                        'account/id',
+                        'account/balances',
+                        'account/history',
+                        'market/orders',
+                        'market/orders/{orderId}',
+                    ],
+                    'post': [
+                        'market/orders',
+                        'market/orders/close/{orderId}',
+                    ],
+                },
+            },
+            'exceptions': {
+                'INVALID ARGUMENT': ExchangeError,
+                'INVALID PARAMETER': ExchangeError,
+                'INSUFFICIENT_FUNDS': InsufficientFunds,
+                'MISSING_API_KEY': AuthenticationError,
+            },
+        });
+    }
+
+    fetchMarkets (params = {}) {
+        let currencies = ['EUR', 'GBP', 'USD', 'CHF', 'PLN'];
+        let markets = [];
+        let currenciesLength = currencies.length;
+        for (let base = 0; base < currenciesLength - 1; base++) {
+            let newIndex = base + 1;
+            for (let quote = newIndex; quote < currencies.length; quote++) {
+                markets.push ({
+                    'id': currencies[base] + '_' + currencies[quote],
+                    'symbol': currencies[base] + '/' + currencies[quote],
+                    'base': currencies[base],
+                    'quote': currencies[quote],
+                    'active': true,
+                    'precision': {
+                        'amount': 2,
+                        'price': 4,
+                    },
+                    'info': currencies[base] + ' - ' + currencies[quote] + ' exchange',
+                });
+            }
+        }
+        return markets;
+    }
+
+    async fetchBalance (params = {}) {
+        let balances = await this.privateGetAccountBalances ();
+        let result = {};
+        for (let i = 0; i < balances.length; i++) {
+            let balance = balances[i];
+            result[this.safeString (balance, 'currency')] = {
+                'free': this.safeFloat (balance, 'balanceAvailable'),
+                'used': this.safeFloat (balance, 'balanceReserved'),
+                'total': this.safeFloat (balance, 'balanceAll'),
+            };
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        let symbolObject = { 'symbol': this.formatSymbol (symbol) };
+        let response = await this.publicGetPublicMarketOrderbookSymbol (this.extend (symbolObject, params));
+        let bids = this.safeValue (response, 'bids', []);
+        let asks = this.safeValue (response, 'asks', []);
+        let result = {
+            'bids': [],
+            'asks': [],
+            'timestamp': undefined,
+            'datetime': undefined,
+            'nonce': undefined,
+        };
+        for (let i = 0; i < 10; i++) {
```

Comment:
I would suggest using `this.parseOrderBook` here instead of `formatOrderBook`+loop ) Can you rewrite it a bit, this will also fix the tiny bug missed in this line that troubles the transpiler. As noted in the contributing guidelines, javascript is ok with referencing assoc keys with `.bids`, but python and php are not, so all assoc keys have to be uniform, in quotes and square brackets (it's technically the same thing in js).

```
result.bids.push  → result['bids'].push
```
↓
```
for (let i = 0...
...
return result;
```

↓

```
return this.parseOrderBook (response, 'bids', 'asks', 'price', 'baseVolume')
```

Reply:
Done :)